analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/MainActivity.kt

```diff
@@ -224,29 +236,29 @@ class MainActivity : AppCompatActivity() {
     }
 
     /**
-     * A [FragmentPagerAdapter] that returns a fragment corresponding to
+     * A [FragmentStateAdapter] that returns a fragment corresponding to
      * one of the sections/tabs/pages.
      */
-    @SuppressLint("WrongConstant")
     inner class ImagePagerAdapter(
         fm: FragmentManager,
-        private val infos: Array<ImageInfo>
-    ) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
+        private val infos: Array<ImageInfo>,
+        lc: Lifecycle
+    ) : FragmentStateAdapter(fm, lc) {
```

Comment:
nit: maybe we should make the variable name consistent with the java sample:
```suggestion
        lifecycle: Lifecycle
    ) : FragmentStateAdapter(fm, lifecycle) {
```